### ISSUE TITLE
[unittest-nightly-autotest] Compilation failure versus dpdk master

### ISSUE THREAD

**Issue by karlatec** [spdk/spdk#1412]
## Link to the failed CI build

https://ci.spdk.io/public_build/autotest-spdk-master-vs-dpdk-master_225.html
https://ci.spdk.io/results/autotest-spdk-master-vs-dpdk-master/builds/225/archive/unittest-nightly-autotest/build.log

## Execution failed at

Error log is pretty long. Attaching it as file [unittest-nightly-autotest.zip](https://github.com/spdk/spdk/files/4661587/unittest-nightly-autotest.zip)
```
0:04:58.112    LINK examples/accel/perf/accel_perf
00:04:58.134  /usr/bin/ld.bfd: /var/jenkins/workspace/unittest-nightly-autotest/dpdk/build/lib/librte_eal.a(eal.o): in function `rte_eal_init.cold':
00:04:58.134  eal.c:(.text.unlikely+0x9d0): undefined reference to `rte_telemetry_init'
00:04:58.134  /usr/bin/ld.bfd: /var/jenkins/workspace/unittest-nightly-autotest/dpdk/build/lib/librte_eal.a(eal_common_options.o): in function `handle_eal_info_request':
00:04:58.134  eal_common_options.c:(.text+0x36): undefined reference to `rte_tel_data_start_array'
00:04:58.134  /usr/bin/ld.bfd: eal_common_options.c:(.text+0x58): undefined reference to `rte_tel_data_add_array_string'
00:04:58.134  /usr/bin/ld.bfd: /var/jenkins/workspace/unittest-nightly-autotest/dpdk/build/lib/librte_eal.a(eal_common_options.o): in function `eal_save_args':
00:04:58.134  eal_common_options.c:(.text+0x821): undefined reference to `rte_telemetry_register_cmd'
00:04:58.134  /usr/bin/ld.bfd: eal_common_options.c:(.text+0x837): undefined reference to `rte_telemetry_register_cmd'
00:04:58.135  collect2: error: ld returned 1 exit status
00:04:58.135  make[3]: *** [/var/jenkins/workspace/unittest-nightly-autotest/spdk/mk/spdk.app.mk:52: ioat_perf] Error 1
00:04:58.136  make[2]: *** [/var/jenkins/workspace/unittest-nightly-autotest/spdk/mk/spdk.subdirs.mk:44: perf] Error 2
00:04:58.136  make[1]: *** [/var/jenkins/workspace/unittest-nightly-autotest/spdk/mk/spdk.subdirs.mk:44: ioat] Error 2
00:04:58.136  make[1]: *** Waiting for unfinished jobs....
00:04:58.148  /usr/bin/ld.bfd: /var/jenkins/workspace/unittest-nightly-autotest/dpdk/build/lib/librte_eal.a(eal.o): in function `rte_eal_init.cold':
00:04:58.148  eal.c:(.text.unlikely+0x9d0): undefined reference to `rte_telemetry_init'
00:04:58.148  /usr/bin/ld.bfd: /var/jenkins/workspace/unittest-nightly-autotest/dpdk/build/lib/librte_eal.a(eal_common_options.o): in function `handle_eal_info_request':
00:04:58.148  eal_common_options.c:(.text+0x36): undefined reference to `rte_tel_data_start_array'
00:04:58.148  /usr/bin/ld.bfd: eal_common_options.c:(.text+0x58): undefined reference to `rte_tel_data_add_array_string'
00:04:58.148  /usr/bin/ld.bfd: /var/jenkins/workspace/unittest-nightly-autotest/dpdk/build/lib/librte_eal.a(eal_common_options.o): in function `eal_save_args':
00:04:58.148  eal_common_options.c:(.text+0x821): undefined reference to `rte_telemetry_register_cmd'
00:04:58.148  /usr/bin/ld.bfd: eal_common_options.c:(.text+0x837): undefined reference to `rte_telemetry_register_cmd'
00:04:58.148  collect2: error: ld returned 1 exit status
00:04:58.149  make[2]: *** [Makefile:46: spdk_trace] Error 1
00:04:58.149  make[1]: *** [/var/jenkins/workspace/unittest-nightly-autotest/spdk/mk/spdk.subdirs.mk:44: trace] Error 2
00:04:58.150  make: *** [/var/jenkins/workspace/unittest-
```

First known failure (SPDK & DPDK head):
```
SPDK:
00:00:12.595  e4c4a0c19 test/make: remove old suppressions.
00:00:12.596  b5eeb44cb test/make: Update ref build path for ABI tests
00:00:12.596  f6fd622bd test/vm_setup.sh: create reference build for ABI tests
00:00:12.596  cb7a83f05 unittest.sh: verify additional dependencies by configure flags
00:00:12.596  9cc916594 doc/jsonrpc: Fix volume size description error

DPDK:
00:00:41.805  24cd1b529 doc: update telemetry guides
00:00:41.805  293c53d8b eal: add telemetry callbacks
00:00:41.806  e122b0bff eal: remove option registration infrastructure
```

Last known good build (SPDK & DPDK head):
```
SPDK:
00:00:14.186  e4c4a0c19 test/make: remove old suppressions.
00:00:14.186  b5eeb44cb test/make: Update ref build path for ABI tests
00:00:14.186  f6fd622bd test/vm_setup.sh: create reference build for ABI tests
00:00:14.186  cb7a83f05 unittest.sh: verify additional dependencies by configure flags
00:00:14.186  9cc916594 doc/jsonrpc: Fix volume size description error

DPDK:
00:00:38.537  dec44d411 eal/x86: add more CPU flags
00:00:38.537  5ebf83784 eal/windows: support logging
00:00:38.538  98e792a35 eal/windows: add fnmatch implementation
```

This issue also affects builds for jobs:
autotest-spdk-v20.01-LTS-vs-dpdk-master
autotest-spdk-v20.04-vs-dpdk-master

**Comment by tmonjalo** [comment]
How SPDK is integrating DPDK?
Is it using pkg-config with libdpdk.pc generated by DPDK meson installation?
If not, I advise switching to pkg-config which is the standard tool for library linkage,
and would provide a robust linkage, independent of any change in DPDK.

**Comment by darsto** [comment]
Hi @tmonjalo . SPDK still uses DPDK Makefiles right now. There are patches to switch to meson, but still in review [1]. Right now SPDK just links with specified rte_* libs and drivers. Switching to pkg-config is something to consider - it might be the way to move forward.

For now I made a quick patch to add rte_telemetry to the lib list: https://review.spdk.io/gerrit/c/spdk/spdk/+/2590

[1] https://review.spdk.io/gerrit/c/spdk/spdk/+/1440

**Comment by tmonjalo** [comment]
Good to know that you may consider using pkg-config. Thanks

**Comment by tomzawadzki** [comment]
[Bug scrub] The patch https://review.spdk.io/gerrit/c/spdk/spdk/+/2590 was merged to SPDK point releases of 20.01 and 20.04, meaning the `vs_dpdk` jobs should not encounter this issue.
For latest SPDK that patch and update to DPDK 20.05 is yet to be done. Along with meson build system update. This is separate task and for that this issue seems resolved.
